src/SelectInput.tsx

```diff
@@ -82,7 +82,12 @@ function SelectInput<V>({
 	const previousItems = useRef<Array<Item<V>>>(items);
 
 	useEffect(() => {
-		if (!isEqual(previousItems.current, items)) {
+		if (
+			!isEqual(
+				previousItems.current.map(item => item.value),
```

Comment:
Hm wait, I think that's incorrect. If `label` changes, list won't be re-rendered with the updated lable. Should it just be `isEqual(previousItems.current, items)`?